.github/workflows/ci.yml

```diff
@@ -0,0 +1,142 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - 'v[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+    branches:
+    - '*'
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v1
+
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install Tox
+      run:  |
+        python -m pip install --upgrade pip
+        pip install tox
+
+    - name: Run Tox
+      run: tox -e py
+      if: ${{matrix.python-version != 3.6 }}
+
+    - name: Run Tox with coverage
+      run: tox -e clean,py
+      if: ${{ matrix.python-version == 3.6 }}
+
+  docs:
+    name: Docs
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+
+    - name: Install Tox
+      run:  |
+        python -m pip install --upgrade pip
+        pip install tox
+
+    - name: Generate Docs
+      run: tox -e docs
+
+  build-and-publish:
```

Comment:
Build and publish step cribbed from various other sources, verified to be working in a separate repository.